iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -92,380 +71,355 @@ TEST_F(ServiceDiscovery_test, FindServiceWithWildcardsReturnsOnlyIntrospectionSe
+    ASSERT_THAT(serviceContainer.value().size(), Eq(1U));
+    EXPECT_THAT(*serviceContainer.value().begin(), Eq(SERVICE_DESCRIPTION));
 
-TEST_F(ServiceDiscovery_test, OfferServiceWithInvalidEventIdFails)
-{
-    auto isServiceOffered = sut.offerService({"service1", iox::capro::InvalidIdString, iox::capro::InvalidIdString});
+    publisher.stopOffer();
     this->InterOpWait();
+    serviceContainer =
+        sut.findService(SERVICE_DESCRIPTION.getServiceIDString(), SERVICE_DESCRIPTION.getInstanceIDString());
+    ASSERT_FALSE(serviceContainer.has_error());
+    EXPECT_TRUE(serviceContainer.value().empty());
 
-    ASSERT_EQ(false, isServiceOffered);
-}
-
-TEST_F(ServiceDiscovery_test, ReofferedServiceWithValidServiceDescriptionCanBeFound)
-{
-    ::testing::Test::RecordProperty("TEST_ID", "b67b4990-e2fd-4efa-ab5d-e53c4ee55972");
-    EXPECT_TRUE(sut.offerService({"service1", "instance1", "event1"}));
-    this->InterOpWait();
-    EXPECT_TRUE(sut.stopOfferService({"service1", "instance1", "event1"}));
+    publisher.offer();
     this->InterOpWait();
-    EXPECT_TRUE(sut.offerService({"service1", "instance1", "event1"}));
-    this->InterOpWait();
-
-    auto serviceContainer = sut.findService(IdString_t("service1"), IdString_t("instance1"));
 
+    serviceContainer =
+        sut.findService(SERVICE_DESCRIPTION.getServiceIDString(), SERVICE_DESCRIPTION.getInstanceIDString());
     ASSERT_FALSE(serviceContainer.has_error());
-    ASSERT_THAT(serviceContainer.value().size(), Eq(1u));
-    ASSERT_THAT(*serviceContainer.value().begin(), Eq(ServiceDescription{"service1", "instance1", "event1"}));
+    ASSERT_THAT(serviceContainer.value().size(), Eq(1U));
+    EXPECT_THAT(*serviceContainer.value().begin(), Eq(SERVICE_DESCRIPTION));
 }
 
 TEST_F(ServiceDiscovery_test, OfferExsistingServiceMultipleTimesIsRedundant)
 {
     ::testing::Test::RecordProperty("TEST_ID", "ae0790ed-4e1b-4f12-94b3-c9e56433c935");
-    EXPECT_TRUE(sut.offerService({"service1", "instance1", "event1"}));
-    this->InterOpWait();
-    EXPECT_TRUE(sut.offerService({"service1", "instance1", "event1"}));
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION("service", "instance", "event");
+    iox::popo::UntypedPublisher publisher(SERVICE_DESCRIPTION);
+    publisher.offer();
     this->InterOpWait();
 
-    auto serviceContainer = sut.findService(IdString_t("service1"), IdString_t("instance1"));
+    auto serviceContainer =
+        sut.findService(SERVICE_DESCRIPTION.getServiceIDString(), SERVICE_DESCRIPTION.getInstanceIDString());
 
     ASSERT_FALSE(serviceContainer.has_error());
-    ASSERT_THAT(serviceContainer.value().size(), Eq(1u));
-    ASSERT_THAT(*serviceContainer.value().begin(), Eq(ServiceDescription{"service1", "instance1", "event1"}));
+    ASSERT_THAT(serviceContainer.value().size(), Eq(1U));
+    EXPECT_THAT(*serviceContainer.value().begin(), Eq(SERVICE_DESCRIPTION));
 }
 
 TEST_F(ServiceDiscovery_test, FindSameServiceMultipleTimesReturnsSingleInstance)
```

Comment:
I'm wondering here, when `ManyToManyPolicy` is used, would a user be interested in the number of offers? This is stored in the `ServiceRegistry` as `ServiceDescriptionEntry::referenceCounter` but currently not passed up to the user API.

Reply:
I was wondering the same in a different way: do we want to pass this information from the registry to the user or ust keep it internal. I see not much relevance from a discovery point of view, either the service exist or it does not. Unless the messages carry more information (in the user data) they are indistinguishable at receiver side, so there is not much point in telling the user that the service is offered multiple times. Certainly not to be changed in this PR.

Reply:
@mossmaurice Good point. I tend to agree with @MatthiasKillat . Since this is not the last PR for the `ServiceDiscovery`, I would suggest to not change this PR and we decide about it in a follow up.

Reply:
@FerdinandSpitzschnueffler Ok, could you please add this to #415 as a todo item?